packages/roosterjs-content-model-api/lib/modelApi/list/setListType.ts

```diff
@@ -121,6 +122,17 @@ export function setListType(
 
                     mutateBlock(parent).blocks.splice(index, 1, newListItem);
                     existingListItems.push(newListItem);
+
+                    const levelIndex = newListItem.levels.length - 1;
+                    const level = mutateBlock(newListItem).levels[levelIndex];
+
+                    if (level) {
+                        updateListMetadata(level, metadata =>
+                            Object.assign({}, metadata, {
+                                applyListStyleFromLevel: true,
```

Comment:
besides here, when do we set this metadata to true?

Reply:
The other place we set this metadata is when indenting a List. In setModelIndentation.ts

Reply:
Ok, I tested, and I did not find any regressions.